### DISCUSSION TITLE
I'm having trouble getting presence to show up on my people search.. what am missing?

### DISCUSSION THREAD

**Discussion by bigbear48858** [microsoft-search/pnp-modern-search#3317] [General]
I have my API permissions on read-all for the presence. I am also added the show-presence to mgt-person section and have the Microsoft graph toolkit turned on..

**Comment by kasperbolarsen** [comment]
And presence is working in ootb people web parts?

**Comment by bigbear48858** [comment]
when I add just a people web part and hover yes it shows the presence

**Comment by bigbear48858** [comment]
![image](https://github.com/microsoft-search/pnp-modern-search/assets/135710111/91cb10ce-ca4d-467b-a0b8-d985bdc558cc)

**Reply by kasperbolarsen** [reply]
Can you share a picture of the config panes?

**Reply by kasperbolarsen** [reply]
Sorry, I didn't mean the People Web part, I was referring to the People Layout in the PnP Modern Search Results web part. Is it working when using that without any modifications?

**Comment by bigbear48858** [comment]
<content id="data-content">

    <style>
        mgt-person {
            --initials-background-color: {{@root.theme.palette.themePrimary}};
        }
.template--defaultList {
	display: flex;
	flex-wrap: wrap;
}

.template--defaultList pnp-select {
	width: 380px;
	height: 110px;
	padding-bottom: 2-px;
}

.ms-Persona-optionalText div {
	display: flex  !important;
	font-size: medium;
}

.ms-Persona-optionalText div {
	padding-right: 10px;
	text-decoration: none;
}

/* unvisited link */
a:link {
  color: black;
}

/* visited link */
a:visited {
  color: black;
}

/* mouse over link */
a:hover {
  color: black;
}

/* selected link */
a:active {
  color: black;
}

    </style>

    <div class="template">

        {{#if @root.properties.showSelectedFilters}}
            <pnp-selectedfilters 
                data-filters="{{JSONstringify filters.selectedFilters 2}}" 
                data-filters-configuration="{{JSONstringify filters.filtersConfiguration 2}}" 
                data-instance-id="{{filters.instanceId}}" 
                data-operator="{{filters.filterOperator}}"
                data-theme-variant="{{JSONstringify @root.theme}}"
            >
            </pnp-selectedfilters>
        {{/if}}
       
        <div class="template--header">
            {{#if @root.properties.showResultsCount}}
                <div class="template--resultCount">
                    <label class="ms-fontWeight-semibold">{{getCountMessage @root.data.totalItemsCount @root.inputQueryText}}</label>
                </div>
            {{/if}}

            <div class="template--sort">
                <pnp-sortfield 
                    data-fields="{{JSONstringify @root.properties.dataSourceProperties.sortList}}" 
                    data-default-selected-field="{{sort.selectedSortFieldName}}" 
                    data-default-direction="{{sort.selectedSortDirection}}"
                    data-theme-variant="{{JSONstringify @root.theme}}">
                </pnp-sortfield>    
            </div>
        </div>
        
        <div>
            <ul class="template--defaultList">
                {{#each data.items as |item|}}
                    <pnp-select 
                        data-enabled="{{@root.properties.itemSelectionProps.allowItemSelection}}" 
                        data-index="{{@index}}" 
                        data-is-selected="{{isItemSelected @root.selectedKeys @index}}">

                        <template id="content">
                            <li class="template--peopleListItem" tabindex="0">
                                {{#> resultTypes item=item}}

                                    {{#and @root.properties.layoutProperties.showPersonaCard (slot item @root.slots.PersonQuery)}}
                                        <mgt-person user-id="{{getUserEmail (slot item @root.slots.PersonQuery)}}"
                                                    show-presence
                                                    view="threeLines"
                                                    person-card="hover">
                                            <template>
                                                <pnp-persona 
                                                    data-image-url="\{{personImage}}" 
                                                    data-fields-configuration="{{JSONstringify @root.properties.layoutProperties.peopleFields}}" 
                                                    data-item="{{JSONstringify item}}" 
                                                    data-persona-size="{{@root.properties.layoutProperties.personaSize}}"
                                                    data-theme-variant="{{JSONstringify @root.theme}}"
                                                    data-instance-id="{{@root.instanceId}}"
                                                    data-context="{{JSONstringify @root}}">
                                                </pnp-persona>
                                            </template>
                                            <template data-type="person-card">
                                                <mgt-person-card inherit-details>
                                                </mgt-person-card>
                                            </template>
                                        </mgt-person>
                                    {{else}}
                                        <pnp-persona 
                                            data-fields-configuration="{{JSONstringify @root.properties.layoutProperties.peopleFields}}"
                                            data-item="{{JSONstringify item}}" 
                                            data-persona-size="{{@root.properties.layoutProperties.personaSize}}" 
                                            data-context="{{JSONstringify @root}}"
                                            data-instance-id="{{@root.instanceId}}"
                                            data-theme-variant="{{JSONstringify @root.theme}}"
                                        ></pnp-persona>
                                    {{/and}}
                                    
                                {{/resultTypes}}
                            </li>
                        </template>
                            
                    </pnp-select>
                {{/each}}
            </ul>
        </div>

        {{#if @root.properties.paging.showPaging}}

            {{#gt @root.data.totalItemsCount @root.properties.paging.itemsCountPerPage}}
                <pnp-pagination 
                    data-total-items="{{@root.data.totalItemsCount}}" 
                    data-hide-first-last-pages="{{@root.properties.paging.hideFirstLastPages}}"
                    data-hide-disabled="{{@root.properties.paging.hideDisabled}}"
                    data-hide-navigation="{{@root.properties.paging.hideNavigation}}"
                    data-range="{{@root.properties.paging.pagingRange}}" 
                    data-items-count-per-page="{{@root.properties.paging.itemsCountPerPage}}" 
                    data-current-page-number="{{@root.paging.currentPageNumber}}"
                    data-theme-variant="{{JSONstringify @root.theme}}"
                >
                </pnp-pagination>
            {{/gt}}
            
        {{/if}}

    </div>

</content>

<content id="placeholder-content">
    <div class="placeholder">

        {{#if @root.properties.showResultsCount}}
            <div class="template--resultCount">
                <span class="placeholder--shimmer placeholder--line" style="width: 20%"></span>
            </div>
        {{/if}}

        <div>
            <ul class="template--defaultList">
                {{#times @root.properties.paging.itemsCountPerPage}}   
                    <pnp-personashimmers 
                        data-persona-size="{{@root.properties.layoutProperties.personaSize}}"
                        data-theme-variant="{{JSONstringify @root.theme}}"
                    >
                    </pnp-personashimmers>
                {{/times}}
            </ul>
        </div>
    </div>
</content>

**Reply by wobba** [reply]
From graph docs it looks like the parameter on `mgt-person` is named `person-presence` not `show-presence`. Could that be it?

**Comment by mopCL** [comment]
Hello,

same issue here...the presence status of Teams is not shown. Neither "show-presence" nor "person-presence" changes anything...any idea?

**Reply by kasperbolarsen** [reply]
All of the graph permissions are granted in the App catalog?

**Reply by mopCL** [reply]
Yes, permissions are granted...

**Reply by till-llit** [reply]
just to provide a "positive" feedback - in our environment it is working. 

we use the details list layout for people search and have the first column configured with this handlebar:

```
<mgt-person 
    user-id="{{item.UserName}}" 
    person-card="hover"
    view="twoLines"
    line2-property="department"
    show-presence>
</mgt-person>
```

presence indicator is visible in the list and the hover popup.

**Reply by mopCL** [reply]
hmm...did a new setup and have the same issue...without any change to the base setup this is what it looks like:

<content id="data-content">

    <style>
        mgt-person {
            --initials-background-color: {{@root.theme.palette.themePrimary}};
        }

    </style>

    <div class="template">

        {{#if @root.properties.showSelectedFilters}}
            <pnp-selectedfilters 
                data-filters="{{JSONstringify filters.selectedFilters 2}}" 
                data-filters-configuration="{{JSONstringify filters.filtersConfiguration 2}}" 
                data-instance-id="{{filters.instanceId}}" 
                data-operator="{{filters.filterOperator}}"
                data-theme-variant="{{JSONstringify @root.theme}}"
            >
            </pnp-selectedfilters>
        {{/if}}
       
        <div class="template--header">
            {{#if @root.properties.showResultsCount}}
                <div class="template--resultCount">
                    <label class="ms-fontWeight-semibold">{{getCountMessage @root.data.totalItemsCount @root.inputQueryText}}</label>
                </div>
            {{/if}}

            <div class="template--sort">
                <pnp-sortfield 
                    data-fields="{{JSONstringify @root.properties.dataSourceProperties.sortList}}" 
                    data-default-selected-field="{{sort.selectedSortFieldName}}" 
                    data-default-direction="{{sort.selectedSortDirection}}"
                    data-theme-variant="{{JSONstringify @root.theme}}">
                </pnp-sortfield>    
            </div>
        </div>
        
        <div>
            <ul class="template--defaultList">
                {{#each data.items as |item|}}
                    <pnp-select 
                        data-enabled="{{@root.properties.itemSelectionProps.allowItemSelection}}" 
                        data-index="{{@index}}" 
                        data-is-selected="{{isItemSelected @root.selectedKeys @index}}">

                        <template id="content">
                            <li class="template--peopleListItem" tabindex="0">
                                {{#> resultTypes item=item}}

                                    {{#and @root.properties.layoutProperties.showPersonaCard (slot item @root.slots.PersonQuery)}}
                                        <mgt-person user-id="{{getUserEmail (slot item @root.slots.PersonQuery)}}" person-card="hover">
                                            <template>
                                                <pnp-persona 
                                                    data-image-url="\{{personImage}}" 
                                                    data-fields-configuration="{{JSONstringify @root.properties.layoutProperties.peopleFields}}" 
                                                    data-item="{{JSONstringify item}}" 
                                                    data-persona-size="{{@root.properties.layoutProperties.personaSize}}"
                                                    data-theme-variant="{{JSONstringify @root.theme}}"
                                                    data-instance-id="{{@root.instanceId}}"
                                                    data-context="{{JSONstringify @root}}">
                                                </pnp-persona>
                                            </template>
                                            <template data-type="person-card">
                                                <mgt-person-card inherit-details>
                                                </mgt-person-card>
                                            </template>
                                        </mgt-person>
                                    {{else}}
                                        <pnp-persona 
                                            data-fields-configuration="{{JSONstringify @root.properties.layoutProperties.peopleFields}}"
                                            data-item="{{JSONstringify item}}" 
                                            data-persona-size="{{@root.properties.layoutProperties.personaSize}}" 
                                            data-context="{{JSONstringify @root}}"
                                            data-instance-id="{{@root.instanceId}}"
                                            data-theme-variant="{{JSONstringify @root.theme}}"
                                        ></pnp-persona>
                                    {{/and}}
                                    
                                {{/resultTypes}}
                            </li>
                        </template>
                            
                    </pnp-select>
                {{/each}}
            </ul>
        </div>

        {{#if @root.properties.paging.showPaging}}

            {{#gt @root.data.totalItemsCount @root.properties.paging.itemsCountPerPage}}
                <pnp-pagination 
                    data-total-items="{{@root.data.totalItemsCount}}" 
                    data-hide-first-last-pages="{{@root.properties.paging.hideFirstLastPages}}"
                    data-hide-disabled="{{@root.properties.paging.hideDisabled}}"
                    data-hide-navigation="{{@root.properties.paging.hideNavigation}}"
                    data-range="{{@root.properties.paging.pagingRange}}" 
                    data-items-count-per-page="{{@root.properties.paging.itemsCountPerPage}}" 
                    data-current-page-number="{{@root.paging.currentPageNumber}}"
                    data-theme-variant="{{JSONstringify @root.theme}}"
                >
                </pnp-pagination>
            {{/gt}}
            
        {{/if}}

    </div>

</content>

<content id="placeholder-content">
    <div class="placeholder">

        {{#if @root.properties.showResultsCount}}
            <div class="template--resultCount">
                <span class="placeholder--shimmer placeholder--line" style="width: 20%"></span>
            </div>
        {{/if}}

        <div>
            <ul class="template--defaultList">
                {{#times @root.properties.paging.itemsCountPerPage}}   
                    <pnp-personashimmers 
                        data-persona-size="{{@root.properties.layoutProperties.personaSize}}"
                        data-theme-variant="{{JSONstringify @root.theme}}"
                    >
                    </pnp-personashimmers>
                {{/times}}
            </ul>
        </div>
    </div>
</content>



-------------------
What setup steps do I miss? I cannot find the corresponding property in the slots nor the properties of the SharePoint search.

Sorry for being this helpless!

**Reply by till-llit** [reply]
look in your template for this line:

` <mgt-person user-id="{{getUserEmail (slot item @root.slots.PersonQuery)}}" person-card="hover">`

you are missing the show-presence parameter. replace it with this line:

` <mgt-person user-id="{{getUserEmail (slot item @root.slots.PersonQuery)}}" person-card="hover" show-presence>`

**Reply by mopCL** [reply]
are you sure with the "" including the presence part?
when I enter that line the whole cards get lost...

**Reply by mopCL** [reply]
and suddenly the hover over card feature does not work any longer...no other changes as suggested...WTF?

**Reply by till-llit** [reply]
my bad, i fixed my code.

**Reply by mopCL** [reply]
after several code copy back and forth I am back with the cards and details on hovering...but no Teams Status/Presence indicator...

**Reply by mopCL** [reply]
sorry to still bother but anyone an idea?

**Reply by mopCL** [reply]
finally works with version 4.11.0...
